src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
```

Comment:
This calc has more complexity than the others, Here's what you need to know to solve it:

1. Use reduce to get the sum value.
1. Get the mother of `children` array. Based on people who has the name equal to the child mother
2. Sum the previous sum to current child born minus the mother born if it exists